internal/cfgplugins/qos.go

```diff
@@ -260,3 +261,90 @@ func enableQosRemarkDscpCliConfig(dut *ondatra.DUTDevice, classifierName string,
 		return ""
 	}
 }
+
+func CreateQueues(t *testing.T, dut *ondatra.DUTDevice, qos *oc.Qos, queues []string) {
+	for index, q := range queues {
+		queue := qos.GetOrCreateQueue(q)
+		queue.Name = ygot.String(q)
+		if deviations.QOSQueueRequiresID(dut) {
+			queue.QueueId = ygot.Uint8(uint8(index))
+		}
+	}
+}
+
+func CreateQueuesFromCli(t *testing.T, dut *ondatra.DUTDevice, queues []string) {
+	cliConfig := ""
+	for index, queue := range queues {
+		cliConfig += fmt.Sprintf(`
+		qos tx-queue %d name %s
+		!`, index, queue)
+	}
+	helpers.GnmiCLIConfig(t, dut, cliConfig)
+}
+
+func configureTwoRateThreeColorSchedulerFromOC(qos *oc.Qos, schedulerName, policerName string, cirValue, pirValue uint64, burstSize, sequenceNumber uint32, queueName string) {
+	sp := qos.GetOrCreateSchedulerPolicy(schedulerName)
+	sp.Name = ygot.String(schedulerName)
+	sched := sp.GetOrCreateScheduler(sequenceNumber)
+	sched.Sequence = ygot.Uint32(1)
+	sched.Type = oc.QosTypes_QOS_SCHEDULER_TYPE_TWO_RATE_THREE_COLOR
+	input := sched.GetOrCreateInput(policerName)
+	input.InputType = oc.Input_InputType_QUEUE
+	input.Queue = ygot.String(queueName)
+	trtc := sched.GetOrCreateTwoRateThreeColor()
+	trtc.Cir = ygot.Uint64(cirValue)
+	trtc.Pir = ygot.Uint64(pirValue)
+	trtc.Bc = ygot.Uint32(burstSize)
+	trtc.Be = ygot.Uint32(burstSize)
+	trtc.GetOrCreateExceedAction().Drop = ygot.Bool(false)
+	trtc.GetOrCreateViolateAction().Drop = ygot.Bool(true)
+}
+
+func configureTwoRateThreeColorSchedulerFromCLI(t *testing.T, dut *ondatra.DUTDevice, schedulerName, className string, cirValue, pirValue uint64, burstSize, queueId uint32) {
+	cliConfig := fmt.Sprintf(`
+policy-map type quality-of-service %s
+   class %s
+   set traffic-class %d 
+   police rate %d bps burst-size %d bytes rate %d bps burst-size %d bytes
+!
+`, schedulerName, className, queueId, cirValue, burstSize, pirValue, burstSize)
+
+	helpers.GnmiCLIConfig(t, dut, cliConfig)
+}
+
+func NewTwoRateThreeColorScheduler(t *testing.T, dut *ondatra.DUTDevice, qos *oc.Qos, schedulerName, policerName, className string, cirValue, pirValue uint64, burstSize, sequenceNumber, queueId uint32, queueName string) {
```

Comment:
each exported cfgplugin function needs to use a struct for all the config parameters and must return a gnmi Batch object.  

See example code at https://github.com/openconfig/featureprofiles/pull/4590/files#diff-6773c4f10a9c00ea10296678db47eb0c295d988ac0d9380074249d2adfec27fcR39-R43